07-data_pipelines.Rmd

```diff
@@ -157,3 +157,74 @@ Link: https://airflow.tools.alpha.mojanalytics.xyz
+
+### Create a test Dag
+
+Using Jupiter lab on the Analytical Platform create a python file in the `airflow/dags` directory in your home directory in the Analytical Platform (eg. `test_dag.py`). Airflow will automatically scan this directory for dag files every three minutes.
+
+See above for creating a Docker image to run the dag if you are using `KubernetesPodOperator`. Concourse will automatically build images of PR's to github airflow repos and will be tagged with the name of the PR.
+
+Example Dag file contents:
+
+
+    from airflow.utils.dates import days_ago
+    from airflow.utils.log.logging_mixin import LoggingMixin
+    from airflow.models import DAG
+
+    log = LoggingMixin().log
+
+    try:
+        from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+
+        args = {"owner": "airflow", "start_date": days_ago(2)}
+
+        dag = DAG(
+            dag_id="example_kubernetes_operator_assume_role",
+            default_args=args,
+            schedule_interval=None,
+        )
+
+        k = KubernetesPodOperator(
+            namespace="user-###YOUR_GITHUB_USERNAME###",
+            image="governmentpaas/awscli:latest",
+            cmds=["aws", "--debug"],
+            arguments=["s3api", "list-objects", "--bucket", "###YOUR_TEST_BUCKET###"],
+            env_vars={
+                "AWS_METADATA_SERVICE_TIMEOUT": "60",
+                "AWS_METADATA_SERVICE_NUM_ATTEMPTS": "5",
+            },
+            labels={"foo": "bar"},
+            name="airflow-test-pod",
+            in_cluster=True,
+            task_id="task",
+            get_logs=True,
+            dag=dag,
+            annotations={"iam.amazonaws.com/role": "alpha_user_###YOUR_GITHUB_USERNAME###"},
+        )
+
+
+    except ImportError as e:
+        log.warn("Could not import KubernetesPodOperator: " + str(e))
+
+
+
+You will have to set the dag namespace to your namespace and the aws role to your role, see `###YOUR_GITHUB_USERNAME###` in the above example code.
+
+You will also need to update the image with your dag image from AWS ECR and add any environment variables etc. as you would with a normal Dag.
+
+### Caveats
+
+This sandbox Airflow is meant for testing. You should not use it to run regular tasks and s should not be relied on for running production pipelines. It is likely that the sandbox will be idled when it is not being used.
+
+Because you are running this Dag as your own `alpha_user_...`, the dag will only have the access that your user has. It does not run with the `iam_policy.json` that you may have created in the task github repo. This means that you might need to either grant yourself access to any buckets or copy data in to a bucket you have created in the Warehouse data section of the Analytical Platform.
```

Comment:
"dag" => "DAG" (sorry :P)

But most important, I'd remove the "only" as it sounds like the DAG will run with not much privileges while that's actually the opposite. My English is not great but we should probably be very very very explicit about the fact that the code will run as their role and potentially do bad things as themselves.

This is my main worry, that because this is a "sandbox"/test airflow they'll not realise that they can do a lot of damage as the code runs like themselves. Don't know if you know what I mean.